arkouda/dataframe.py

```diff
@@ -1841,6 +1844,269 @@ def numeric_help(d):
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        if not self.name:
+            raise RegistrationError(
+                "This DataFrame does not have a name and does not appear to be registered."
+            )
+
+        DataFrame.unregister_dataframe_by_name(self.name)
+
+        self.name = None  # Clear our internal DataFrame object name
+
+    def is_registered(self) -> bool:
+        """
+        Return True if the object is contained in the registry
+
+        Returns
+        -------
+        bool
+            Indicates if the object is contained in the registry
+
+        Raises
+        ------
+        RegistrationError
+            Raised if there's a server-side error or a mismatch of registered components
+
+        See Also
+        --------
+        register, attach, unregister, unregister_dataframe_by_name
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        if self.name is None:
+            return False
+
+        # Total number of registered parts should equal number of columns plus an entry
+        # for the index and the column order
+        total = len(self.data) + 2
+        registered = 0
+
+        for col, data in self.data.items():
+            if data.is_registered():
+                registered += 1
+
+        if self.index.values.is_registered():
+            registered += 1
+        if f"df_columns_{self.name}" in list_registry():
+            registered += 1
+
+        if 0 < registered < total:
+            warn(
+                f"WARNING: DataFrame {self.name} expected {total} components to be registered,"
+                f" but only located {registered}."
+            )
+            return False
+        else:
+            return registered == total
+
+    @staticmethod
+    def attach(user_defined_name: str) -> DataFrame:
+        """
+        Function to return a DataFrame object attached to the registered name in the
+        arkouda server which was registered using register()
+
+        Parameters
+        ----------
+        user_defined_name : str
+            user defined name which DataFrame object was registered under
+
+        Returns
+        -------
+        DataFrame
+               The DataFrame object created by re-attaching to the corresponding server components
+
+        Raises
+        ------
+        RegistrationError
+            if user_defined_name is not registered
+
+        See Also
+        --------
+        register, is_registered, unregister, unregister_groupby_by_name
+        """
+
+        registry = list_registry()
+
+        col_resp = cast(str, generic_msg(cmd="attach", args=f"df_columns_{user_defined_name}"))
+        cols = Strings.from_return_msg(col_resp)
+        columns = dict.fromkeys(cols.to_list())
```

Comment:
Can we not just return a list of columns from chapel? Since we do not need the strings object here, we are doing more than is required. Something along the lines of 
```chapel
var x: [0..#3] string = ["abc", "123", "def"];
writeln("%jt".format(x));
```
This way the python can just run `json.loads` instead of creating a strings object and then calling `to_list`.

Reply:
Implemented `stringsToJSONMsg` return to enable this functionality